### ISSUE TITLE
RFC 2616 and Webservice

### ISSUE THREAD

**Issue by Max84** [PrestaShop/PrestaShop#16720]
**Describe the bug**
RFC 2616 tells http headers are case insensitive and should be in lower-case.
I suggest to change PSWS-Version variable to psws-version because some load balancers and proxies convert http headers to lowercase and programming language are case sensitive. It can break some development

**To Reproduce**
Just open the webservice API and check header to see it in camel case.

**Additional information**
PrestaShop version: all
PHP version: all

**Comment by Max84** [comment]
In fact RFC 2616 was remplaced by RFC 7231 see here: https://www.rfc-editor.org/rfc/rfc7231.txt

And this is an extract :

`RFC 7231             HTTP/1.1 Semantics and Content            June 2014


   The type, subtype, and parameter name tokens are case-insensitive.
   Parameter values might or might not be case-sensitive, depending on
   the semantics of the parameter name.  The presence or absence of a
   parameter might be significant to the processing of a media-type,
   depending on its definition within the media type registry.

   A parameter value that matches the token production can be
   transmitted either as a token or within a quoted-string.  The quoted
   and unquoted values are equivalent.  For example, the following
   examples are all equivalent, but the first is preferred for
   consistency:

     text/html;charset=utf-8
     text/html;charset=UTF-8
     Text/HTML;Charset="utf-8"
     text/html; charset="utf-8"`

**Comment by khouloudbelguith** [comment]
Hi @Max84,

>Parameter values might or might not be case-sensitive

So, the parameter value should in the minimal case as described in this screenshot?
![image](https://user-images.githubusercontent.com/16067358/70321202-02eb7d00-1827-11ea-8f0c-c4bd87eee72d.png)

Thanks!

**Comment by Max84** [comment]
Not at all, your fields must be case sensitive.
I only talk about the http header "PSWS-Version" sent by webservice. You can see it with an html inspector (with Firefox, maybe Chrome).

![image](https://user-images.githubusercontent.com/6622400/70326238-57e1c000-1834-11ea-93e9-95283e78fef5.png)

Edit: a better screenshot

**Comment by Max84** [comment]
You've add the "bug" tag. For me, it's not really a bug, but an improvement.
I don't know if it is important for you.

**Comment by khouloudbelguith** [comment]
@Max84, thanks for these clarifications.
Ping @PrestaShop/prestashop-product-team what do you think?

Thanks!

**Comment by eternoendless** [comment]
As far as I can tell, there is no recommendation for lower case headers in both RFCs. They only refer to their _interpretation_, which as you cited, should be case-insensitive.

As you said, programming languages are case sensitive. Therefore it's the clients (and not the server) who must be adapted to comply with RFCs and interpret headers in a case-insensitive way, or be subject to potential problems involving intermediate layers which could alter the HTTP header's case.

Moreover, changing the PSWS-version header to lowercase would be a breaking change, in terms that non RFC compliant clients will certainly fail to interpret said header because of case sensitiveness.

To recap:

- If a client implements the RFCs correctly, there is zero gain in applying the requested change, because it doesn't care about case.
- If a client doesn't implement RFCs correctly, this change will break it, because it expects a camel cased header.

Therefore, I'm not favorable to this change.
